### ISSUE TITLE
Creation of shaded dataflow-nullaway artifact is broken

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#7398]
When I run `./gradlew publishToMavenLocal`, a jar file `~/.m2/repository/org/checkerframework/dataflow-nullaway/3.52.1-SNAPSHOT/dataflow-nullaway-3.52.1-SNAPSHOT.jar` is created.  But when I list the files in that jar they are incorrect:
```bash
$ jar tf dataflow-nullaway-3.52.1-SNAPSHOT.jar | head -15
META-INF/MANIFEST.MF
META-INF/LICENSE.txt
MANIFEST.MF
org/checkerframework/shaded/dataflow/analysis/AbstractAnalysis$Worklist$BackwardDfoComparator.class
org/checkerframework/shaded/dataflow/analysis/AbstractAnalysis$Worklist$ForwardDfoComparator.class
org/checkerframework/shaded/dataflow/analysis/AbstractAnalysis$Worklist.class
org/checkerframework/shaded/dataflow/analysis/AbstractAnalysis.class
org/checkerframework/shaded/dataflow/analysis/AbstractValue.class
org/checkerframework/shaded/dataflow/analysis/Analysis$BeforeOrAfter.class
org/checkerframework/shaded/dataflow/analysis/Analysis$Direction.class
org/checkerframework/shaded/dataflow/analysis/Analysis.class
org/checkerframework/shaded/dataflow/analysis/AnalysisResult$1.class
org/checkerframework/shaded/dataflow/analysis/AnalysisResult.class
org/checkerframework/shaded/dataflow/analysis/BackwardAnalysis.class
org/checkerframework/shaded/dataflow/analysis/BackwardAnalysisImpl$1.class
```
The classes are under `org/checkerframework/shaded` when they should be under `org/checkerframework/nullaway`.  NullAway is currently using version 3.49.2 so this must have broken sometime after that.

I tried understanding better what is going on by building the jar:
```bash
$ ./gradlew clean :dataflow:dataflownullawayJar
[succeeds]
$ ls dataflow/build/shadow/dataflownullaway/
dataflow-errorprone-3.52.1-SNAPSHOT.jar
```
The jar file has the wrong name, but when I check its contents they are as expected (classes under `org/checkerframework/nullaway`).  This seems to be some issue with build configuration and maybe files getting overwritten?  But I don't quite understand it.  Unfortunately, I'm blocked on testing out changes like https://github.com/typetools/checker-framework/pull/7397 within the context of NullAway until this gets fixed.

**Comment by msridhar** [comment]
I just checked on Maven Central, and the jar for version 3.49.5 is fine, while the jar for 3.50.0 suffers from the above issue.

**Comment by msridhar** [comment]
Sorry, #7402 fixed the issue with `./gradlew clean :dataflow:dataflownullawayJar` producing an incorrectly-named jar, but it did _not_ fix the issue with the published jars.  Even on the latest master branch with #7402 included I still see the shaded classes for `dataflow-nullaway` ending up under `org/checkerframework/shaded` when I do `./gradlew publishToMavenLocal`, same for `dataflow-errorprone`.  So @smillst seems like we still need to do something else there.
